src/generic_actor.cr

```diff
@@ -13,15 +13,17 @@ module GenericActor
   class AbuseStoppedActorException < Exception
   end
 
-  @message_loop_state = ActorState::PENDING
-  @message_loop_state_mutex = Mutex.new
+  @message_loop_state = Atomic(ActorState).new(ActorState::PENDING)
```

Comment:
Actually if param is enum you  can use Symbol to pass value, it is made transparent by compiler.
From my point of view it is more readable than fully qualified names  `ActorState::PENDING` -> ':pending'

Reply:
And, is that compile time checkable for typo errors?

Reply:
yes, exactly, because of Symbols can't be defined in runtime.